core/trino-main/src/main/java/io/trino/execution/scheduler/DynamicSplitPlacementPolicy.java

```diff
@@ -36,7 +35,7 @@ public DynamicSplitPlacementPolicy(NodeSelector nodeSelector, Supplier<? extends
     }
 
     @Override
-    public SplitPlacementResult computeAssignments(Set<Split> splits)
+    public SplitPlacementResult computeAssignments(List<Split> splits)
```

Comment:
> Convert Split to record

I'm not sure if we really want to change Split to record causing the collection type change. I would leave to other reviewers.

Reply:
Records have generated equals and hashCode which is causing issues if you put them into a set. I've discussed that with @losipiuk and @findepi. It seems thar Sets where used just for the sake of tests

Reply:
cc @martint @dain

Reply:
I can extract this change to a separate PR

Reply:
Extracted to https://github.com/trinodb/trino/pull/21583. I'll drop this commit here